src/components/Form/FormField.tsx

```diff
@@ -0,0 +1,221 @@
+    required: PropTypes.bool,
+
+    /** Text message that will be displayed below the control (can be used for error, warning, success messages). */
+    message: customPropTypes.itemShorthand,
+
+    /**
+     * A custom render function for the message slot.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderMessage: PropTypes.func,
+
+    /** Additional CSS styles to apply to the component instance.  */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Override for theme site variables to allow modifications of component styling via themes. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+  }
+
+  public static defaultProps = {
+    as: 'div',
+  }
+
+  public renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    variables,
+    styles,
+    rest,
+  }): React.ReactNode {
+    const {
+      children,
+      control,
+      renderControl,
+      label,
+      renderLabel,
+      id,
+      type,
+      required,
+      message,
+      renderMessage,
+      controlType,
+    } = this.props
+
+    let labelContent = label
+
+    if (required) {
+      labelContent =
+        typeof label === 'string'
+          ? `${label} *`
+          : typeof label === 'object' && (label as any).content
+            ? { ...label, content: `${(label as any).content} *` }
+            : label
+    }
+
+    const labelElement = Text.create(labelContent, {
+      defaultProps: {
+        as: 'label',
+        htmlFor: id,
+        styles: styles.label,
+      },
+      render: renderLabel,
+    })
+
+    const messageElement = Text.create(message, {
+      defaultProps: {
+        styles: styles.message,
+      },
+      render: renderMessage,
+    })
+
+    const factoryMethod = createSlotFactory(controlType, name => ({ name }))
+    const controlElement = factoryMethod(control || {}, {
+      defaultProps: { required, ...rest },
+      render: renderControl,
+    })
+
+    // ----------------------------------------
+    // No Control
+    // ----------------------------------------
+
+    if (!controlType) {
+      if (!label) {
+        return (
+          <ElementType {...rest} className={classes.root}>
+            {childrenExist(children) && children}
+            {messageElement}
+          </ElementType>
+        )
+      }
+
+      return (
+        <ElementType {...rest} className={classes.root}>
+          {labelElement}
+          {messageElement}
+        </ElementType>
+      )
+    }
+
+    // --------------------------------------------------------
+    // Check box or radio (label should appear after the input)
+    // --------------------------------------------------------
+
+    if (controlType && (type === 'checkbox' || type === 'radio')) {
```

Comment:
`controlType` check not needed
```suggestion
    if (type === 'checkbox' || type === 'radio') {
```